wire-ios-sync-engine/Source/Synchronization/SyncAgent.swift

```diff
@@ -280,6 +291,51 @@ final class SyncAgent: NSObject, SyncAgentProtocol {
                 self?.resume()
             }
     }
+
+    private func handlePushChannelAlreadyOpened() async {
+        WireLogger.sync.debug("push channel opened, waiting until closed", attributes: .syncAttributes)
+        await pushChannelMonitor.waitUntilPushChannelClosed()
+        WireLogger.sync.debug("retry sync after NSE push channel closed", attributes: .syncAttributes)
+        resume()
+    }
+}
+
+private extension PushChannelMonitorProtocol {
+
+    func waitUntilPushChannelClosed() async {
+        await withCheckedContinuation { continuation in
+            var resumed = false
+
+            Task {
+                // in case the other process is killed we resume anyway
+                // so we're not stuck
+                do {
+                    try await Task.sleep(for: .seconds(5))
+                    guard !resumed else { return }
+                    resumed = true
+                    WireLogger.sync.debug(
+                        "timed out waiting for push channel to be closed",
```

Comment:
suggestion(non-blocking): I think it'd be nice to move the timeout logic inside the monitor.